cpu/atmega_common/periph/rtc.c

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (C) 2018 Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_atmega_common
+ * @ingroup     drivers_periph_rtc
+ *
+ * @{
+ *
+ * @file
+ * @brief       RTC interface wrapper for use with RTT modules.
+ *
+ * @author      Matthew Blue <matthew.blue.neuro@gmail.com>
+ *
+ * @}
+ */
+
+#include <avr/interrupt.h>
+#include <time.h>
+
+#include "cpu.h"
+#include "periph/rtc.h"
+#include "periph/rtt.h"
+#include "periph_conf.h"
+#include "thread.h"
+
+#ifdef MODULE_XTIMER
+#include "xtimer.h"
+#endif
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+/* guard file in case no RTC device is defined */
+#if RTC_NUMOF
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct {
+    time_t time;          /* seconds since the epoch */
+    time_t alarm;         /* alarm_cb when time == alarm */
+    rtt_cb_t alarm_cb;    /* callback called from RTC alarm */
+    void *alarm_arg;      /* argument passed to the callback */
+} rtc_state_t;
+
+static volatile rtc_state_t rtc_state;
+
+void rtc_init(void)
+{
+    /* Initialize rtc_state*/
+    rtc_state.time = 0;
+    rtc_state.alarm = 0;
+    rtc_state.alarm_cb = NULL;
+    rtc_state.alarm_arg = NULL;
+
+    /* RTC depends on RTT */
+    rtt_init();
+}
+
+int rtc_set_time(struct tm *time)
+{
+    /* Convert to seconds since the epoch */
+    rtc_state.time = mktime(time);
+
+    return 0;
+}
+
+int rtc_get_time(struct tm *time)
+{
+    /* Convert from seconds since the epoch */
+    /* Note: Cast tells the compiler to discard volatile */
+    gmtime_r((time_t*)&rtc_state.time, time);
+
+    return 0;
+}
+
+int rtc_set_alarm(struct tm *time, rtc_alarm_cb_t cb, void *arg)
+{
+    /* Disable alarm interrupt */
+    TIMSK2 &= ~(1 << OCIE2B);
+
+    /* Set callback */
+    rtc_state.alarm_cb = cb;
+    rtc_state.alarm_arg = arg;
+
+    /* Wait until not busy anymore (should be immediate) */
+    DEBUG("RTC sleeps until safe to write OCR2B\n");
+    while( ASSR & (1 << OCR2BUB) ) {
+#ifdef MODULE_XTIMER
```

Comment:
I suggest factoring this ifdef into it's own static function (maybe with the umber of RTC cycles as parameter), it is used at least twice.

Reply:
Done. Also done for RTT.